Remove `unicorn/filename-case` because it's not working as we thought it would

get-off-my-lawn actually allows kebab and pascal case, [as configured here](https://github.com/manovotny/eslint-config-get-off-my-lawn/blob/master/src/unicorn.js#L6-L14). However, this is the error it's producing.

```bash
/Users/manovotny/Developer/lerna-yarn-workspaces-monorepo/packages/button/src/Button.spec.js
  1:1  error  Filename is not in kebab case or pascal case. Rename it to `button.spec.js` or `Button.Spec.js`  unicorn/filename-case
```

What we want is actually a mix of the two, ie. `Button.spec.js`, which isn't really allowed.

[I've mentioned this on the unicorn project](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/203#issuecomment-526004462).

We probably should remove the rule until it can accomplish what we want.

--- a/src/unicorn.js
+++ b/src/unicorn.js
@@ -3,15 +3,6 @@ module.exports = {
     'unicorn/custom-error-definition': 'error',
     'unicorn/error-message': 'error',
     'unicorn/escape-case': 'error',
-    'unicorn/filename-case': [
-        'error',
-        {
-            cases: {
-                kebabCase: true,
-                pascalCase: true
-            }
-        }
-    ],
     'unicorn/new-for-builtins': 'error',
     'unicorn/no-abusive-eslint-disable': 'error',
     'unicorn/no-array-instanceof': 'error',

